### DISCUSSION TITLE
Troubles Librewolf on Fedora

### DISCUSSION THREAD

**Discussion by xdettlaff** [filips123/PWAsForFirefox#323] [Q&A] [answered]
Hello, plz help to use it with librewolf on fedora:

I created 
└[$]> mkdir -p $HOME/.local/share/firefoxpwa/runtime/ 

└[$]> ln librewolf firefox
ln: there is no 'librewolf': No file or directory

so i do 
└[$]> ln -s /usr/share/librewolf/librewolf firefox

then i downloaded librewolf tar
![image](https://user-images.githubusercontent.com/113319266/232526734-7ad75ef8-108f-4202-8b34-2adb3f305173.png)

> Extract downloaded file and copy content (without "root" directory) of librewolf-VERSION\LibreWolf\ to %APPDATA%\FirefoxPWA\runtime\.

i dont see a "root" here, wt you a talking about?
so i just copy it like that 
![image](https://user-images.githubusercontent.com/113319266/232528228-40955619-df16-4ded-9c93-4423f4db5b99.png)

but it doesnt work...

**Comment by filips123** [comment]
> ln: there is no 'librewolf': No file or directory

You should run that after you've extracted LibreWolf to the `$HOME/.local/share/firefoxpwa/runtime/` directory. 

> i dont see a "root" here, wt you a talking about?

I meant the root directory of the archive, not the directory actually named "root". I see that this is confusing and will improve the wording in the future.

> so i just copy it like that

This seems fine, now you should just run that `ln` from before (from inside that directory):

```bash
ln librewolf firefox
```

**Reply by xdettlaff** [reply]
thanks, i run it without errors
![image](https://user-images.githubusercontent.com/113319266/232571127-97e63966-10ac-4213-a7a0-00d1dba427b9.png)

now i have 
![image](https://user-images.githubusercontent.com/113319266/232571220-85f64bda-66d3-46cd-bd0f-8ecf875732d6.png)

and i have installed 
![image](https://user-images.githubusercontent.com/113319266/232571373-67484a31-a449-4b3a-b48a-65ea0ea79d7d.png)

but i still have this error 
![image](https://user-images.githubusercontent.com/113319266/232571502-6b5dcd9e-bc13-4e62-b326-3985425d82c1.png)

**Reply by filips123** [reply]
Are you also using LibreWolf as a main browser? Then, you also have to run these two commands so LibreWolf finds native app manifests installed for normal Firefox:

```bash
ln -s ~/.mozilla/native-messaging-hosts ~/.librewolf/native-messaging-hosts
sudo ln -s /usr/lib/mozilla/native-messaging-hosts /usr/lib/librewolf/native-messaging-hosts
```

Alternatively, you can copy `/usr/lib/mozilla/native-messaging-hosts/firefoxpwa.json` to `/usr/lib/librewolf/native-messaging-hosts/firefoxpwa.json`?

**Comment by xdettlaff** [comment]
└[$]> ln -s ~/.mozilla/native-messaging-hosts ~/.librewolf/native-messaging-hosts 
ln: failed to create symbolic link '/home/def/.librewolf/native-messaging-hosts': File exists

└[$]> sudo ln -s /usr/lib/mozilla/native-messaging-hosts /usr/lib/librewolf/native-messaging-hosts
ln: failed to create symbolic link '/usr/lib/librewolf/native-messaging-hosts': No such file or directory

I'm sorry if I'm stupid :((((

**Reply by filips123** [reply]
Ok, then I think it would be easier if you just copy file `/usr/lib/mozilla/native-messaging-hosts/firefoxpwa.json` into `~/.librewolf/native-messaging-hosts`. Hopefully that will work.

Sorry, you are not stupid, but the documentation definitely needs to be improved. I'm working on writing a new version, but haven't finished it yet.

**Reply by xdettlaff** [reply]
sorry, i dont understand what u mean, can you give me some commands? do i need to move file... or create new.... im not a firefox developer :(

i had installed firefox (remove it now), so this file ```/usr/lib/mozilla/native-messaging-hosts/firefoxpwa.json ```
already exists in my system

![image](https://user-images.githubusercontent.com/113319266/233141861-7f974afc-8106-4b38-b72b-c3a2bc5bbfd5.png)

**Reply by filips123** [reply]
Copy `/usr/lib/mozilla/native-messaging-hosts/firefoxpwa.json` into `~/.librewolf/native-messaging-hosts`:

```bash
mkdir -p ~/.librewolf/native-messaging-hosts # In case it does not already exist
cp /usr/lib/mozilla/native-messaging-hosts/firefoxpwa.json ~/.librewolf/native-messaging-hosts/
```

**Reply by xdettlaff** [reply]
> Copy `/usr/lib/mozilla/native-messaging-hosts/firefoxpwa.json` into `~/.librewolf/native-messaging-hosts`:
> 
> ```shell
> mkdir -p ~/.librewolf/native-messaging-hosts # In case it does not already exist
> cp /usr/lib/mozilla/native-messaging-hosts/firefoxpwa.json ~/.librewolf/native-messaging-hosts/
> ```


```
└[$]> mkdir -p ~/.librewolf/native-messaging-hosts # In case it does not already exist
mkdir: The directory "/home/def/.librewolf/native-messaging-hosts" cannot be created: File exists

└[$]> cp /usr/lib/mozilla/native-messaging-hosts/firefoxpwa.json ~/.librewolf/native-messaging-hosts/
cp: Cannot create normal file '/home/def/.librewolf/native-messaging-hosts/': This is not the directory

└[$]> cp /usr/lib/mozilla/native-messaging-hosts/firefoxpwa.json ~/.librewolf/native-messaging-hosts/firefoxpwa.json 
cp: Cannot create normal file '/home/def/.librewolf/native-messaging-hosts/firefoxpwa.json': No such file or directory
```

/native-messaging-hosts" cannot be created: File exists     - this is file, soo.... your commands didnt work

**Reply by filips123** [reply, marked as answer]
You can try to remove file `~/.librewolf/native-messaging-hosts` and run the above commands again.

**Reply by xdettlaff** [reply]
thanks! it works now, you need to add this commands to wiki ^-^
